README.md

```diff
@@ -1,3 +1,437 @@
+
+**Prerequisite**: MacOS version must be **9.0** or higher.
+
+Open a new terminal window.
+
+Navigate to the previously downloaded folder that contains the source code of the DID project.
+
+```shell
+$ cd YOUR-PATH/Elastos.DID.Native.SDK
+```
+
+Enter the 'build' folder.
+
+```shell
+$ cd build
+```
+
+Create a new folder with the target platform name, then change directory.
+
+```shell
+$ mkdir ios
+$ cd ios
+```
+
+To generate the required Makefile in the current directory, please make sure to first replace 'YOUR-IOS-PLATFORM' with the correct option.
+
+-DIOS_PLATFORM accepts the following target architecture options:
+* iphoneos
+* iphonesimulator
+
+Replace 'YOUR-IOS-PLATFORM' with the path to the extracted NDK folder.
+
+Run the command with the correct options described above:
+
+```shell
+$ cmake -DIOS_PLATFORM=YOUR-IOS-PLATFORM -DCMAKE_TOOLCHAIN_FILE=../../cmake/iOSToolchain.cmake ../..
+
+```
+
+Build the program:
+
+*Note: If "make" fails due to missing permissions, use "sudo make" instead.*
+
+```shell
+$ make
+```
+
+Install the program:
+
+*Note: If "make install" fails due to missing permissions, use "sudo make install" instead.*
+
+```shell
+$ make install
+```
+
+Create distribution package:
+
+*Note: If "make dist" fails due to missing permissions, use "sudo make dist" instead.*
+
+```shell
+$ make dist
+```
+
+## Build on Windows Host
+
+COMING SOON
+
+## Build API Documentation
+
+Currently, the API documentation can only be built on **Linux** hosts. MacOS has a bug issue with python, which would cause build process failure.
+
+### Build on Ubuntu / Debian / Linux Host
+
+#### 1. Install Pre-Requirements
+
+```shell
+$ sudo apt-get update
+$ sudo apt-get install doxygen python-sphinx graphviz
+$ curl -L -o /tmp/get-pip.py https://bootstrap.pypa.io/get-pip.py
+$ sudo python /tmp/get-pip.py
+$ sudo pip install breathe
+```
+
+#### 2. Build
+
+Change to the directory where the build for any target has been previously executed. For example, if the target was Linux, the folder structure would be similar to:
+
+```shell
+cd /YOUR-PATH/Elastos.DID.Native.SDK/build/linux
+```
+
+Run the following command:
+
+*Note: If "make" fails due to missing permissions, use "sudo make" instead.*
+
+```shell
+$ cmake -DENABLE_DOCS=ON ../..
+$ make
+```
+
+#### 3. View
+
+The generated API documentation will be created in the new **/docs** directory on the same directory level.
+
+Change to the docs folder:
+
+```shell
+cd docs/html
+```
+Open the index.html file in a browser from the terminal:
+
+```shell
+xdg-open index.html
+```
+
+## Contribution
+
+We welcome contributions to the Elastos DID Project.
+
+## Acknowledgments
+
+A sincere thank you to all teams and projects that we rely on directly or indirectly.
+
+## License
+
+This project is licensed under the terms of the [GPLv3 license](https://github.com/elastos/Elastos.NET.Carrier.Native.SDK/blob/master/LICENSE).
```

Comment:
Better to use MIT instead of GPLv3, which for Carrier Native SDK, it has to be GPLV3 because of it's dependency library toxcore using GPLv3 .